povm_toolbox/post_processor/median_of_means.py

```diff
@@ -0,0 +1,196 @@
+        self, observable: SparsePauliOp, loc: int | tuple[int, ...]
+    ) -> tuple[float, float]:
+        count = self.counts[loc]
+        shots = sum(count.values())
+        omegas = self.get_decomposition_weights(observable, set(count.keys()))
+
+        # get the samples and randomize their order
+        sampled_weights = np.zeros(shots)
+        idx = 0
+        for outcome in count:
+            sampled_weights[idx : idx + count[outcome]] = omegas[outcome]
+            idx += count[outcome]
+        sampled_weights = self.seed.permutation(sampled_weights)
+
+        # compute the batch size
+        batch_size = shots // self.num_batches
+        batches = sampled_weights[: self.num_batches * batch_size].reshape(
+            (batch_size, self.num_batches)
+        )
+
+        # if the numbers of shots is not a multiple of number of batches, some batches
+        # will have ``batch_size`` samples and some will have ``batch_size+1`` samples
+        # such that all samples are used.
+        if shots % self.num_batches != 0:
+            last_samples = np.full((1, self.num_batches), np.nan)
+            last_samples[:, : shots % self.num_batches] = sampled_weights[
+                self.num_batches * batch_size :
+            ]
+            batches = np.concatenate((batches, last_samples))
+
+        # compute the median of means estimate
+        median_of_means = float(np.median(np.nanmean(batches, axis=0)))
+
+        # compute the coefficient in front of the shadow norm in the formula to compute
+        # the confidence parameter ``delta``.
+        coefficient_epsilon = float(np.sqrt(34 / batch_size))
+
+        return median_of_means, coefficient_epsilon
+
+    def get_expectation_value(
+        self,
+        observable: SparsePauliOp,
+        loc: int | tuple[int, ...] | None = None,
+    ) -> tuple[np.ndarray, np.ndarray] | tuple[float, float]:
+        r"""Return the expectation value and confidence parameter of the given ``observable``.
+
+        Args:
+            observable: the observable whose expectation value is queried.
+            loc: this argument is relevant if multiple sets of parameter values were supplied to the
+                sampler in the same :class:`.POVMSamplerPub`. The index ``loc`` then corresponds to
+                the set of parameter values that was supplied to the sampler through the Pub. If
+                ``None``, the expectation value (and standard deviation) for each set of circuit
+                parameters is returned.
+
+        Returns:
+            A tuple of (estimated) expectation value(s) and pre-coefficient(s) for the confidence
+            parameter :math:`\epsilon`, that is, :math:`\sqrt{34 / \mathrm{batch\_size} }`.
+            If a single value was queried (via ``loc``), both of these will be a ``float``.
+            Otherwise, they will be instances of :class:`numpy.ndarray`.
+        """
+        return super().get_expectation_value(observable, loc)
```

Comment:
I don't think we need this. The explanations of the class docstring should be sufficiently clear that the user understands that the output of this method will be computed in a different way.